src/ClientData/include/ClientData/TrackData.h

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CLIENT_DATA_TRACK_DATA_H_
+#define CLIENT_DATA_TRACK_DATA_H_
+
+#include <absl/synchronization/mutex.h>
+
+#include "ClientData/TimerChain.h"
+
+namespace orbit_client_data {
+
+class TrackData {
+ public:
+  [[nodiscard]] size_t GetNumberOfTimers() const { return num_timers_; }
+  [[nodiscard]] uint64_t GetMinTime() const { return min_time_; }
+  [[nodiscard]] uint64_t GetMaxTime() const { return max_time_; }
+
+  TextBox& AddTimer(uint64_t depth, orbit_client_protos::TimerInfo timer_info) {
+    auto* timer_chain = GetOrCreateTimerChain(depth);
+    UpdateMinTime(timer_info.start());
+    UpdateMaxTime(timer_info.end());
+    ++num_timers_;
+
+    return timer_chain->emplace_back(std::move(timer_info));
+  }
+
+  [[nodiscard]] std::vector<TimerChain*> GetChains() {
+    std::vector<TimerChain*> chains;
+    absl::MutexLock lock(&mutex_);
+    for (auto& it : timers_) {
+      chains.push_back(it.second.get());
+    }
+
+    return chains;
+  }
+
+  [[nodiscard]] TimerChain* GetChain(uint64_t depth) {
```

Comment:
we cannot return const TimerChain here because there are parts of code that modify returned TextBox. And this is a reason why iterator cannot be made const. I do not like it either but before the code was basically lying about being const where it was not really const. I decided to make it explicit. My hope is that once we remove TextBox we can restore constness here properly - by making timer chain iterator const as well, like it was done for BlockChain

Reply:
done.